src/main/java/com/minecolonies/coremod/entity/CustomArrowEntity.java

```diff
@@ -83,6 +90,20 @@ protected void onHitEntity(EntityRayTraceResult traceResult)
 
         // Set the old actual damage value back
         setBaseDamage(prevDamage);
+        if (onHitCallback != null && onHitCallback.test(traceResult))
```

Comment:
Kinda find it weird if a predicate has a side-effect, maybe make it a producer or a function?

Reply:
its the same as a function, it just doesnt need unboxing